extensions/hibernate-search-orm-elasticsearch/runtime/src/main/java/io/quarkus/hibernate/search/orm/elasticsearch/runtime/HibernateSearchElasticsearchBuildTimeConfig.java

```diff
@@ -22,30 +25,50 @@ public class HibernateSearchElasticsearchBuildTimeConfig {
      *
      * @asciidoclet
      */
-    @ConfigItem(defaultValue = "true")
-    public boolean enabled;
+    @WithDefault("true")
+    boolean enabled();
 
     /**
      * Configuration for the default persistence unit.
      */
-    @ConfigItem(name = ConfigItem.PARENT)
-    public HibernateSearchElasticsearchBuildTimeConfigPersistenceUnit defaultPersistenceUnit;
+    @WithParentName
+    HibernateSearchElasticsearchBuildTimeConfigPersistenceUnit defaultPersistenceUnit();
 
     /**
      * Configuration for additional named persistence units.
      */
     @ConfigDocSection
     @ConfigDocMapKey("persistence-unit-name")
-    @ConfigItem(name = ConfigItem.PARENT)
-    Map<String, HibernateSearchElasticsearchBuildTimeConfigPersistenceUnit> persistenceUnits;
+    @WithParentName
+    Map<String, HibernateSearchElasticsearchBuildTimeConfigPersistenceUnit> persistenceUnits();
 
-    public Map<String, HibernateSearchElasticsearchBuildTimeConfigPersistenceUnit> getAllPersistenceUnitConfigsAsMap() {
+    default Map<String, HibernateSearchElasticsearchBuildTimeConfigPersistenceUnit> getAllPersistenceUnitConfigsAsMap() {
         Map<String, HibernateSearchElasticsearchBuildTimeConfigPersistenceUnit> map = new TreeMap<>();
+        HibernateSearchElasticsearchBuildTimeConfigPersistenceUnit defaultPersistenceUnit = defaultPersistenceUnit();
+
         if (defaultPersistenceUnit != null) {
             map.put(PersistenceUnitUtil.DEFAULT_PERSISTENCE_UNIT_NAME, defaultPersistenceUnit);
         }
-        map.putAll(persistenceUnits);
+        map.putAll(persistenceUnits());
         return map;
     }
 
+    default HibernateSearchElasticsearchBuildTimeConfigPersistenceUnit getPersistenceUnitConfig(String persistenceUnitName) {
```

Comment:
In other places we just do `getPersistenceUnitConfigsAsMap().get(persistenceUnitName)`... which seems simpler than having a dedicated method.


Ideally we'd introduce a dedicated construct in the config mapping code, and we wouldn't have to reproduce those utils everywhere... E.g. instead of:

```java
    @WithParentName
    HibernateSearchElasticsearchBuildTimeConfigPersistenceUnit defaultPersistenceUnit();
    @WithParentName
    Map<String, HibernateSearchElasticsearchBuildTimeConfigPersistenceUnit> persistenceUnits();

    default Map<String, HibernateSearchElasticsearchBuildTimeConfigPersistenceUnit> getAllPersistenceUnitConfigsAsMap() {
        // ...
    }

    default HibernateSearchElasticsearchBuildTimeConfigPersistenceUnit getPersistenceUnitConfig(String persistenceUnitName) {
        // ...
    }
```

We'd simply do this:

```java
    @WithParentName
    @WithRootEntry(key = PersistenceUnitUtil.DEFAULT_PERSISTENCE_UNIT_NAME)
    Map<String, HibernateSearchElasticsearchBuildTimeConfigPersistenceUnit> persistenceUnits();
```

Would be nice to introduce this before we migrate all extensions... But I imagine we don't really have time for that? My understanding was that we were going to migrate to `@ConfigMapping` *after* 3.0.0.Final, but I guess that changed.

Reply:
We can avoid creating the method if you prefer. I wanted to be extra safe but I agree the checks should be useless.

Reply:
I'd personally favor the simpler option, yes.

I created #32295 to perhaps address this directly in the config extension.

Reply:
I kept this for now because it's practical, we can drop it at some later point.